packages/core/lib/generators/assets_generator.dart

```diff
@@ -29,7 +30,7 @@ class AssetsGenConfig {
 
   factory AssetsGenConfig.fromConfig(File pubspecFile, Config config) {
     return AssetsGenConfig._(
-      pubspecFile.parent.path,
+      pubspecFile.parent.absolute.path,
```

Comment:
We believe the problem can be solved by simply passing the RootPath to isSupport. No big changes to assets_generator.dart is needed.